internal/cloudapi/server.go

```diff
@@ -254,12 +254,22 @@ func (server *Server) ComposeStatus(w http.ResponseWriter, r *http.Request, id s
 		return
 	}
 
+	var convertTargetResults []TargetResult
+	var tr TargetResult
+
+	for _, x := range result.TargetResults {
+		tr.Name = x.Name
+		tr.Options = x.Options
+		convertTargetResults = append(convertTargetResults, tr)
+	}
+
 	response := ComposeStatus{
 		ImageStatus: ImageStatus{
 			Status: composeStatusFromJobStatus(status, &result),
 			UploadStatus: &UploadStatus{
-				Status: result.UploadStatus,
-				Type:   "aws",
+				Status:        result.UploadStatus,
+				Type:          "aws",
```

Comment:
Keeping `"aws"` value hard-coded here is not beneficial in my opinion, especially once we introduce additional targets. The type of the target is now part of the `TargetResults` items as their `Name` property. Therefor this information is redundant. This is related to my other comment I left in the `internal/cloudapi/openapi.yml` file.

Reply:
The type needs to be extracted from the TargetResult into the UploadStatus exactly. But I think we need to make a decision first on if we're going to support multiple upload_requests per compose or not.

Either we support multiple upload_requests and then we need to have an array in UploadStatus. Or we don't and then this can be a single object.

Reply:
My thought process with `UploadStatus` and `TargetResults` was that these are technically the same thing. The target is basically an upload + additional processing needed to be able to use the image for VM instances. So the overall upload status does not makes so much sense, because with multiple targets you would be interested of the result of each one of them (basically the upload status). The same goes for errors and additional target-specific results (AMI ID, etc...).

Based on the discussion I had so far about this part of composer, I think that we still want to keep the ability to have multiple targets, even if not really allowed to be requested via cloudapi for now.

Since `ComposeRequest` -> `ImageRequest`'s `upload_requests` property is an array of `UploadRequest` I would propose to make the `ImageStatus` structure in the Compose status response symmetric. I would propose to keep only the `UploadStatus` as you mentioned and not expose the `TargetResults`. Also because the naming seems to be more of an internal thing, than anything we use in the cloudapi. In addition I would make the `upload_status` property of `ComposeStatus` an array. It should contain an item for each `UploadRequest`.